diff --git a/engine/liberty.h b/engine/liberty.h
index 0bddff5..8a83c1c 100644
--- a/engine/liberty.h
+++ b/engine/liberty.h
@@ -857,8 +857,8 @@ struct worm_data {
 extern struct worm_data worm[BOARDMAX];
 
 /* Unconditionally meaningless moves. */
-int meaningless_black_moves[BOARDMAX];
-int meaningless_white_moves[BOARDMAX];
+static int meaningless_black_moves[BOARDMAX];
+static int meaningless_white_moves[BOARDMAX];
 
 /* Surround cache (see surround.c) */
 
